### PR TITLE
Update tensor.py

### PR DESCRIPTION
**Pull request by whalmcfly**
Fixing "_one_hot_along_dim input needs to be int #9179" caused incorrect behavior in the "sparse_categorical_crossentropy" function.
- Added a cast to int32.
- Uncertain whether similar changes should be applied elsewhere in tensor.py.

### CODE REVIEW

**Comment by chenyuxyz** [comment]
what are you trying to fix?

**Comment by whalmcfly** [comment]
I was running:
"def step():
    """Perform a single optimization step."""
    Tensor.training = True  # makes dropout work
    samples = Tensor.randint(BATCH_SIZE, high=X_train.shape[0])
    x_train_batch, y_train_batch = X_train[samples], Y_train[samples]

    optim.zero_grad()
    labels, _ = model(x_train_batch.float())
    labels = labels.float()
    y_train_batch = y_train_batch.float()
    loss = labels.sparse_categorical_crossentropy(y_train_batch)
    loss.backward()

    optim.step()
    return loss

jit_step = TinyJit(step)"

and sparse_categorical_crossentropy gave me an error that it takes only integers. Maybe I'm wrong.

**Comment by chenyuxyz** [comment]
the error is saying don't convert y_train_batch to float before passing into sparse_categorical_crossentropy
